lib/ChGovUk/Plugins/FormHelpers.pm

```diff
@@ -484,18 +495,23 @@ sub _get_errors_for_field {
             }
 
             $error = Locale::Simple::l($error, { %error_args } );
-            my $error_id = $self->_field_name_to_id($field) . '-' . $rule . '-error';
-            push @errors, { text => $error, id => $error_id };
 
             # Add error to global error array
             my %error_hash;
             $error_hash{name} = $field;
             $error_hash{text} = $error;
 
-            # Does the controller have a corresponding field?
-            if ( defined $self->labels->{$field} ) {
-                $error_hash{id}   = $error_id;
-                push $self->errors, \%error_hash;
+            if ($self->_is_warning($rule)) {
+                push $self->warnings, \%error_hash;
```

Comment:
Although this is possible - probably safest to dereference this:

`push @{$self->warnings}, \%error_hash;
`

for more info: https://perldoc.perl.org/functions/push.html

Reply:
Done!